src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -473,9 +487,33 @@ async def _determine_requirements_for_pex_from_targets(
     return dataclasses.replace(requirements, from_superset=repository_pex.maybe_pex), ()
 
 
+def _warn_about_any_files_targets(
+    addresses: Addresses, transitive_targets: TransitiveTargets, union_membership: UnionMembership
+) -> None:
+    # Warn if users depend on `files` targets, which won't be included in the PEX and is a common
+    # gotcha.
+    file_tgts = targets_with_sources_types(
+        [FileSourceField], transitive_targets.dependencies, union_membership
+    )
+    if file_tgts:
+        files_addresses = sorted(tgt.address.spec for tgt in file_tgts)
+        formatted_addresses = ", ".join(str(a) for a in addresses)
+        targets, depend = ("target", "depends") if len(addresses) == 1 else ("targets", "depend")
+        logger.warning(
+            f"The {targets} {formatted_addresses} transitively {depend} "
```

Comment:
This text is slightly worse, as it has lost the target alias name foo/bar:baz ..`` , and just mentions the address:

- before: ``The `pex_binary` target foo/bar:baz transitively depends ...``
- after: ``The target foo/bar:baz transitively depends ...``

Is it possible to get more info about a target from an address somehow? E.g. get back to the original `Target` subclass or get its `Alias`?

Reply:
You can use something like
```python
targets = await Get(Targets, Addresses, addresses)
```
https://www.pantsbuild.org/docs/rules-api-and-target-api#how-to-resolve-targets

Reply:
Thanks, done. Not exactly the same as it was (to handle the possibility of multiple addresses/targets with less special casing), but still covers the intention, I think?